src/main/java/com/nvidia/spark/rapids/jni/HyperLogLogPlusPlusHostUDF.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.jni;
+
+import ai.rapids.cudf.ColumnVector;
+import ai.rapids.cudf.ColumnView;
+import ai.rapids.cudf.NativeDepsLoader;
+
+/**
+ * HyperLogLogPlusPlus(HLLPP) host UDF aggregation utils
+ */
+public class HyperLogLogPlusPlusHostUDF {
+  static {
+    NativeDepsLoader.loadNativeDeps();
+  }
+
+  /**
+   * HyperLogLogPlusPlus(HLLPP) aggregation types
+   */
+  public enum AggregationType {
+
+    /**
+     * Compute hash codes for the input, generate HyperLogLogPlusPlus(HLLPP)
+     * sketches from hash codes, and merge all the sketches into one sketch, output
+     * is a struct scalar with multiple long values.
+     */
+    Reduction(0),
+
+    /**
+     * Merge all HyperLogLogPlusPlus(HLLPP) sketches in the input column into one
+     * sketch. Input is a struct column with multiple long columns which is
+     * consistent with Spark. Output is a struct scalar with multiple long values.
+     */
+    Reduction_MERGE(1),
```

Comment:
Naming convention should be consistent with `GroupByMerge`.
```suggestion
    ReductionMerge(1),
```

Reply:
done